FAQs:[CDS-93887]: FAQs for 22nd March 24 (#6042)
harness/developer-hub: kb/continuous-delivery/continuous-delivery-faqs.md

   Replace `/path/to/new/helm/binary` with the actual path to the newly installed Helm binary.

3. Restart the delegate to apply the changes.

#### Which entities such as service or environment are factors that determine the metrics displayed in Deployment Dashboard?

In our setup, two Looker dashboard models are specifically designed to showcase data solely from pipeline executions with a CD stage. The data aggregation and presentation within these views adapt dynamically based on the chosen attributes for display on the dashboard
Please read more on Looker/Dashboard Behaviour in the following [Doumentation](https://developer.harness.io/docs/cloud-cost-management/use-ccm-cost-reporting/ccm-cost-categories/cost-categories-usage/#behaviour)

> #### Is `Custom Stage` considered as Continuous Delivery stage? Is execution of a pipeline without any Service entity considered towards Total deployments?

Reviewer: ```suggestion
#### Is `Custom Stage` considered a Continuous Delivery stage? Is execution of a pipeline without any Service entity considered towards Total deployments?
```

Yes, we consider Custom stage as part of CD stage as of now to update the dashboards, and a deployments without Service is not possible for CD Stage hence it would not be considered towards total deployments
Please read more on custom stages in the following [Documentation](https://developer.harness.io/docs/platform/pipelines/add-a-stage/#add-a-custom-stage)

#### What is the correct expression to reference artifact version in rollback phase?

One can use the expression `rollbackArtifact.version` . This is a change from FirstGen to NextGen where rollback artifact version was getting automatically resolved

> Please find an example usecase on this in our [Documentation](https://developer.harness.io/docs/continuous-delivery/deploy-srv-diff-platforms/custom-deployment-tutorial/#pipeline-sample-setup---cicd)

Reviewer: ```suggestion
Please find an example use case on this in our [Documentation](https://developer.harness.io/docs/continuous-delivery/deploy-srv-diff-platforms/custom-deployment-tutorial/#pipeline-sample-setup---cicd)
```

#### Does Harness allow license key, and how is it used?

> Harness refer API keys instead of license keys to enter in product.

Reviewer: ```suggestion
Harness prefer API keys instead of license keys to enter in product.
```

Please read more on this in the following [Documentation](https://developer.harness.io/docs/platform/automation/api/add-and-manage-api-keys/)

#### Does Harness provide control over Abort Pipeline to users exclusively?

> Yes, Harness provides control over abort pipelines as feature behind FF: `CDS_PIPELINE_ABORT_RBAC_PERMISSION`. This feature allows control abort operation that kills the pipeline/stage without any cleanup

Reviewer: ```suggestion
Yes, Harness provides control over abort pipelines as a feature behind the FF: `CDS_PIPELINE_ABORT_RBAC_PERMISSION`. This feature allows a control abort operation that kills the pipeline/stage without any cleanup.
```

> This consists pre-requisite FF: `CDS_PIPELINE_ABORT_RBAC_PERMISSION_MIGRATION` enabled required for this feature in use, if this is not done, one may wind up with users who could abort the pipeline earlier and now cannot.

Reviewer: ```suggestion
This consists of a pre-requisite FF: `CDS_PIPELINE_ABORT_RBAC_PERMISSION_MIGRATION` enabled required for this feature to be in use, if this is not done, one may wind up with users who could abort the pipeline earlier and now cannot.
```
